### ISSUE TITLE
ezdxf.lldxf.const.DXFTableEntryError: OpenSansCondensed-Light and EZDXF

### ISSUE THREAD

**Issue by benediktbrandt** [mozman/ezdxf#121]
I just downloaded the latest version of ezdxf (v0.10.4) to test the fix for https://github.com/mozman/ezdxf/issues/9

Unfortunately I am now getting an `ezdxf.lldxf.const.DXFTableEntryError: OpenSansCondensed-Light` error during DXF generation for the following piece of code:

```
dim = self.msp.add_linear_dim(base=(offset_x, -self.vertical_padding + dimension_line_padding),
                                                  p1=dimension_top_points[j], p2=dimension_top_points[j + 1],
                                                  text=dimension_top_points_labels[j], angle=0, dimstyle='arrow',
                                                  dxfattribs={'layer': dimension_layer_name})
dim.render()
```

where `arrow` is a custom defined dimstyle. Here is the corresponding stacktrace

```
File "/home/benedikt/anaconda3/envs/image-recognition-pyms2/lib/python3.6/site-packages/ezdxf/entities/dimstyleoverride.py", line 447, in render
    renderer = self.get_renderer(ucs)
  File "/home/benedikt/anaconda3/envs/image-recognition-pyms2/lib/python3.6/site-packages/ezdxf/entities/dimstyleoverride.py", line 425, in get_renderer
    return self.doc.dimension_renderer.dispatch(self, ucs)
  File "/home/benedikt/anaconda3/envs/image-recognition-pyms2/lib/python3.6/site-packages/ezdxf/render/dimension.py", line 1169, in dispatch
    return self.linear(dimension, ucs, override)
  File "/home/benedikt/anaconda3/envs/image-recognition-pyms2/lib/python3.6/site-packages/ezdxf/render/dimension.py", line 1187, in linear
    return LinearDimension(dimension, ucs, override)
  File "/home/benedikt/anaconda3/envs/image-recognition-pyms2/lib/python3.6/site-packages/ezdxf/render/dimension.py", line 707, in __init__
    super().__init__(dimension, ucs, override)
  File "/home/benedikt/anaconda3/envs/image-recognition-pyms2/lib/python3.6/site-packages/ezdxf/render/dimension.py", line 127, in __init__
    self.text_style = self.drawing.styles.get(self.text_style_name)  # type: Style
  File "/home/benedikt/anaconda3/envs/image-recognition-pyms2/lib/python3.6/site-packages/ezdxf/sections/table.py", line 116, in get
    raise DXFTableEntryError(name)
ezdxf.lldxf.const.DXFTableEntryError: OpenSansCondensed-Light
```

If I omit `dimstyle='arrow'` from the `add_linear_dim` call I get another (similar) error: `ezdxf.lldxf.const.DXFTableEntryError: EZDXF`. Here is the corresponding stack trace.

```
File "/home/benedikt/anaconda3/envs/image-recognition-pyms2/lib/python3.6/site-packages/ezdxf/graphicsfactory.py", line 937, in add_linear_dim
    style = DimStyleOverride(dimline, override=override)
  File "/home/benedikt/anaconda3/envs/image-recognition-pyms2/lib/python3.6/site-packages/ezdxf/entities/dimstyleoverride.py", line 19, in __init__
    self.dimstyle = self.doc.dimstyles.get(dim_style_name)  # type: DimStyle
  File "/home/benedikt/anaconda3/envs/image-recognition-pyms2/lib/python3.6/site-packages/ezdxf/sections/table.py", line 116, in get
    raise DXFTableEntryError(name)
ezdxf.lldxf.const.DXFTableEntryError: EZDXF
```

I tried with DXF version DXF2000 and DXF2018, but the error is the same. Interestingly if I specify the following setup options (which I originally added to solve https://github.com/mozman/ezdxf/issues/9) the error goes away (for both DXF2000 and DXF2018). 

```
self.doc = ezdxf.new(DXF2000, setup=['linetypes', 'styles', 'dimstyles'])
```

Please let me know if you need any more information.

**Comment by mozman** [comment]
It seems `ezdxf.new(setup=True)` solves all your problems, why not using it?

**Comment by mozman** [comment]
If you define you own dimstyle, be sure the used text style is defined, `OpenSansCondensed-Light` is the default style, but not defined if don't use setup:

    doc.dimstyles.new('arrow', dxfattribs={'dimtxsty': 'xyz'})
    assert 'xyz' in doc.styles

or use `ezdxf.new(setup=True)`

If you are using the default dimstyles `EZDXF`, be sure the dimstyle is defined:

    assert 'EZDXF' in doc.dimstyles

or use `ezdxf.new(setup=True)`

**Comment by benediktbrandt** [comment]
@mozman thanks for your replies and information. Can you elaborate on the differences between `ezdxf.new(setup=True)` and `ezdxf.new()` and under what circumstances one should use one over the other? 

I ran a few tests:

 ezdxf.new(DXF2000, setup=True): No error messages form ezdxf but copy to clipboard fails in Autocad 2020 (presumably same issue as in https://github.com/mozman/ezdxf/issues/9)
 ezdxf.new(DXF2018, setup=True): No error messages form ezdxf but copy to clipboard fails in Autocad 2020 
ezdxf.new(DXF2000): `ezdxf.lldxf.const.DXFTableEntryError: OpenSansCondensed-Light`
ezdxf.new(DXF2018): `ezdxf.lldxf.const.DXFTableEntryError: OpenSansCondensed-Light`
ezdxf.new(DXF2000, setup=['linetypes', 'styles', 'dimstyles']): No error messages from ezdxf and copy and pasting works in Autocad 2020 

When we define our dimstyle we are not specifying `dimtxsty` so it is using the default value. Here is the full code snippet:

```
self.doc.dimstyles.new(name="arrow", dxfattribs={'dimtxt': dimension_line_font_height,
                                                        'dimblk': ezdxf.ARROWS.closed_filled,  # set arrow style
                                                        'dimtsz': 0,  # set tick size to 0, to enable arrow usage
                                                        'dimasz': dimension_line_font_height,
                                                        # arrow size in drawing units
                                                        'dimclrt': 60,  # color
                                                        'dimclre': 60,  # color
                                                        'dimclrd': 60,  # color
                                                        'dimexe': dimension_line_font_height * 0.5,
                                                        # length above dimension line
                                                        'dimexo': dimension_line_font_height * 0.2,
                                                        # offset from measurement point
                                                        'dimgap': dimension_line_font_height * 0.33,
                                                        })
```

But I have seen the above behavior with and without our own dimstyle so I doubt it is related. 

I am happy to do some testing in AutoCAD 2020 if need be.

**Comment by mozman** [comment]
`setup=True` just setup all implemented standards, it is the same as `setup =['linetypes', 'styles', 'visualstyles', 'dimstyles']`. see [source code](https://github.com/mozman/ezdxf/blob/b7327131bd9c14267e2c14dd49ee11b8cfa85227/src/ezdxf/tools/standards.py#L17)

Errors without `setup` are clear, this is changed in the next release, none existing text and dimension styles will be replaced by the default `Standard` styles, but with the disadvantage of no feedback for mistyped styles, only a debug log will be created, but only user who setup the `logging` module will see the message.

If you are using the patched version v0.10.3 or later and the copy to clipboard error still occurs, I can't really help, because I don't know the reason and the DXF reference is not helpful for such errors, it is the DXF reference and not "What AutoCAD expects from a DXF file" documentation - which is not the same.

The only thing you could do, is to run the `RECOVER` command in AutoCAD and upload the output here or send it to ezdxf@mozman.at including a "simplified" Python code which creates such a DXF file, please add version info about ezdxf and the Python interpreter.

Don't have to much hope, it was just luck to find such error at the first time, and this kind of error has no high priority to me because you can fix this by saving the file as `.dwg` file. If you have the problem of copying entities to clipboard you have already launched AutoCAD and saving the file as `.dwg` is not a great effort.
